satellite/contact/endpoint.go

```diff
@@ -42,23 +49,23 @@ func (endpoint *Endpoint) CheckIn(ctx context.Context, req *pb.CheckInRequest) (
 
 	peerID, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.Unknown, errCheckInIdentity.New("failed to get ID from context: %v", err).Error())
 	}
 	nodeID := peerID.ID
 
 	err = endpoint.service.peerIDs.Set(ctx, nodeID, peerID)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.Unknown, errCheckInIdentity.New("failed to get ID from context: %v", err).Error())
 	}
 
 	lastIP, err := overlay.GetNetwork(ctx, req.Address)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.NotFound, errCheckInNetwork.New("failed to get IP from address: %s, err: %v", req.Address, err).Error())
 	}
 
 	pingNodeSuccess, pingErrorMessage, err := endpoint.pingBack(ctx, req, nodeID)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.NotFound, errCheckInNetwork.New("failed to ping back address:%s, err: %v", req.Address, err).Error())
```

Comment:
also, it might be nice to differentiate between some of the failure modes that pingBack has. there's a dialing error, a connect error, an internal error, and a communincation error in there.